File: athena/src/app/model/Aula.ts
import {Disciplina} from "./Disciplina";
import {Atividade} from "./Atividade";

export interface Aula{
  "id": number
  "titulo": string
  "tema": string
  "link_video": string,
  "disciplina": Disciplina,
  atividades: Atividade[]
}

File: athena/src/main/java/br/com/athena/model/Aula.java
package br.com.athena.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.*;
import java.util.List;

@Entity
public class Aula {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String titulo;

    private String tema;

    @JsonProperty("link_video")
    private String linkVideo;

    @ManyToOne
    private Disciplina disciplina;

    @OneToMany(mappedBy = "aula", cascade = CascadeType.ALL)
    @JsonIgnoreProperties("aula")
    private List<Atividade> atividades;

}
